contracts/SnappAuctionChallenge.sol

```diff
@@ -36,46 +36,43 @@ contract SnappAuctionChallenge {
         uint24 rolloverCount;
     }
 
+    modifier onlyVerifiedOrdersAndSolution(bytes memory pricesAndVolumes, bytes memory orders) {
+        require(checkPriceAndVolumeData(pricesAndVolumes), "Wrong prices or volumes");
+        require(checkOrderData(orders), "Wrong order data");
+        _;
+    }
+
     /**
      * Top level challenges
      */
     function challengePriceNonUniform(
         bytes memory pricesAndVolumes,
         bytes memory orders,
         uint16 badOrder
-    ) public returns (bool) {
-        require(!checkPriceAndVolumeData(pricesAndVolumes), "Wrong prices or volumes");
-        require(!checkOrderData(orders), "Wrong order data");
-        
+    ) public view onlyVerifiedOrdersAndSolution(pricesAndVolumes, orders) returns (bool) {
         Order memory order = getOrder(orders, badOrder);
         (uint buyVolume, uint sellVolume) = getVolumes(pricesAndVolumes, badOrder);
         uint buyPrice = getPrice(pricesAndVolumes, order.buyToken);
         uint sellPrice = getPrice(pricesAndVolumes, order.sellToken);
 
-        require(challengePriceNonUniform(buyPrice, sellPrice, buyVolume, sellVolume), "Prices are uniform");
-        return true;
+        require(priceNotUniform(buyPrice, sellPrice, buyVolume, sellVolume), "Prices are uniform");
+        return true;  // Why not just return priceNotUniform?
```

Comment:
@fleupold ---!

Reply:
again, to make the tests pass and get an estimate on a full succesful verification without doing all the bit-calculation to make the prices actually uniform.